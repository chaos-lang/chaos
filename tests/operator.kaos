print "Relational operators on boolean variables"
bool a = true
bool b = false
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on integer variables"
num a = 3
num b = 5
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on float variables"
num a = 13.5
num b = 763.78
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on a mix of integer and boolean variables"
bool a = true
num b = 37
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on a mix of integer and float variables"
num a = 98
num b = 643.134
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on a mix of float and boolean variables"
bool a = false
num b = 3.14
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on a mix of boolean variables and boolean/integer/float values"
bool a = true
bool b = false
bool c = a == 431
print c
del c
bool c = 3.14 != b
print c
del c
bool c = false > b
print c
del c
bool c = a < true
print c
del c
bool c = 1.0 >= 15
print c
del c
bool c = 14114 <= 345435.3414
print c
del c
del a
del b


print "Logical operators on boolean variables"
bool a = true
bool b = false
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on integer variables"
num a = 3
num b = 5
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on float variables"
num a = 13.5
num b = 763.78
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on a mix of integer and boolean variables"
bool a = true
num b = 37
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on a mix of integer and float variables"
num a = 98
num b = 643.134
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Relational operators on a mix of boolean variables and boolean/integer/float values"
bool a = true
bool b = false
bool c = a && 431
print c
del c
bool c = 3.14 and b
print c
del c
bool c = false || b
print c
del c
bool c = a or true
print c
del c
bool c = ! 15
print c
del c
bool c = not 345435.3414
print c
del c
del a
del b


print "Bitwise operators"
num a = 5
num b = 9
num c = a & b
print c
del c
num c = a | b
print c
del c
num c = a ^ b
print c
del c
num c = ~ a
print c
del c
num c = b << 1
print c
del c
num c = b >> 1
print c
del c
del a
del b

print "Unary operators"
num a = 3
num b = 5
num c = (a + b) * a
print c
del a
del b
del c

num a = (3 + 5) * 3
print a
del a

num a = 3
num b = ++a
print b
del b
num b = a++
print b
del b
num b = --a
print b
del b
num b = a--
print a
del a
