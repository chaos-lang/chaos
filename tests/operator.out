Relational operators on boolean variables
false
true
true
false
true
false
Relational operators on integer variables
true
false
false
false
true
false
Relational operators on float variables
true
false
false
false
true
false
Relational operators on a mix of integer and boolean variables
true
false
false
false
true
false
Relational operators on a mix of integer and float variables
true
false
false
false
true
false
Relational operators on a mix of float and boolean variables
false
true
false
true
false
true
Relational operators on a mix of boolean variables and boolean/integer/float values
false
true
false
false
false
true
Logical operators on boolean variables
false
false
true
true
false
false
Logical operators on integer variables
true
true
true
true
false
false
Logical operators on float variables
true
true
true
true
false
false
Logical operators on a mix of integer and boolean variables
true
true
true
true
false
false
Logical operators on a mix of integer and float variables
true
true
true
true
false
false
Relational operators on a mix of boolean variables and boolean/integer/float values
true
false
false
true
false
false
Bitwise operators
1
13
12
-6
18
4
Unary operators
24
24
4
4
4
3
